### PR TITLE
[crashpad] Fix build errors under linux

### PR DESCRIPTION
**Pull request by jimwang118**
Fix one of https://github.com/microsoft/vcpkg/issues/32398 issue.
```
src/951b70517d-a4e19cb5ec.clean/third_party/mini_chromium/mini_chromium/base/atomicops_internals_portable.h:57:25: error: cannot initialize a parameter of type 'int' with an lvalue of type 'const memory_order'
   57 |   __atomic_thread_fence(std::memory_order_seq_cst);
```
Modify issue based on upstream [bug](https://issues.chromium.org/issues/42310321) fix. 
Since there is no Clang++ compiler on the CI linux machine, the compilation fails, so I can't remove crashpad:x64-linux=fail from ci.baseline.txt.

- [X] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md).
- [ ] ~~SHA512s are updated for each updated download.~~
- [ ] ~~The "supports" clause reflects platforms that may be fixed by this new version.~~
- [ ] ~~Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.~~
- [ ] ~~Any patches that are no longer applied are deleted from the port's directory.~~
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is added to each modified port's versions file.

Usage test pass with following triplets:

```
x64-windows
x64-linux
```

### CODE REVIEW

**Comment by RealChuan** [comment]
https://github.com/chromium/mini_chromium/commit/b870d45e275006f9996f807c16df36d81b346b26
https://github.com/chromium/mini_chromium/commit/b870d45e275006f9996f807c16df36d81b346b26
This is an upstream update, it may be better to update crashpad and mini_chromium versions

**Comment by jimwang118** [comment]
> [chromium/crashpad@da18935](https://github.com/chromium/crashpad/commit/da189353b60611dd7089aa2a14441893460ebcdb) [chromium/mini_chromium@b870d45](https://github.com/chromium/mini_chromium/commit/b870d45e275006f9996f807c16df36d81b346b26) This is an upstream update, it may be better to update crashpad and mini_chromium versions

Thanks for your suggestion, it has been modified.

File: ports/crashpad/portfile.cmake [hunk 1]
```diff
@@ -4,12 +4,14 @@ vcpkg_from_git(
...
     REF 7e0af1d4d45b526f01677e74a56f4a951b70517d
+    PATCHES
+        fix-linux.patch
 )
 
 vcpkg_find_acquire_program(PYTHON3)
 vcpkg_replace_string("${SOURCE_PATH}/.gn" "script_executable = \"python3\"" "script_executable = \"${PYTHON3}\"")
 
-function(checkout_in_path PATH URL REF)
+function(checkout_in_path PATH URL REF PAT)
```
Comment (dg0yt) [inline ports/crashpad/portfile.cmake:14]: Bad interface. Works for a single patch. Requires `""` if no patch is needed.

Comment (jimwang118) [reply ports/crashpad/portfile.cmake]: It has been fixed.

**Review by Cheney-W — APPROVED** [review]

**Pull request merged**